### DISCUSSION TITLE
Read a variable in a C# custom activity

### DISCUSSION THREAD

**Discussion by johnvarney612** [elsa-workflows/elsa-core#5716] [Q&A]
How do I set a variable in an Http request activity and then read the variable in a C# custom activity using the designer?

**Comment by glime-ai** [comment]
To set a variable in an HTTP request activity and then read it in a C# custom activity using the designer in Elsa Workflows, you can follow these steps:

1. **Create the Variable:**
   - In the designer, use the Variables panel to create a new variable that will hold the value you want to pass between activities. Give it a name and select the appropriate type.

2. **Set the Variable in the HTTP Request Activity:**
   - Configure your HTTP request activity to store the desired data into the variable you created. You can do this by setting the output properties of the HTTP request activity to bind to your variable.

3. **Read the Variable in the Custom Activity:**
   - In your custom C# activity, you can read the variable using the `ActivityExecutionContext` provided in the `Execute` method. Use the `GetVariable` method to retrieve the value of the variable by its name.

Here's an example of how you might retrieve a variable named "MyVariable" within an activity:

```csharp
protected override async ValueTask<IActivityExecutionResult> OnExecuteAsync(ActivityExecutionContext context)
{
    // Retrieve the variable value from the workflow instance storage.
    var myVariableValue = context.GetVariable<string>("MyVariable");

    // Use the variable value as needed in your activity logic.
    // ...

    return Done();
}
```

In this example, `GetVariable<T>` is a generic method where `T` is the type of the variable you are retrieving. Replace `string` with the appropriate data type of your variable.

Please note that for this to work, the variable "MyVariable" must have been previously created and stored in the workflow instance. This can be done programmatically when building the workflow or through the designer interface.

For more information on variables and their usage in Elsa, you can refer to the documentation on variables at the provided link.

If you need further clarification or assistance, please let me know!

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
